meadow-client/src/Simulation.purs

```diff
@@ -401,6 +412,47 @@ transactionButtons state = [ div [ classes [ ClassName "d-flex"
                                    ]
                            ]
 
+printTransWarnings :: forall p. BigInteger -> List DynamicProblem -> Array (HTML p Query)
```

Comment:
Because Int overflows. I know it is unlikely and something else will probably crash first, but I don't see the need to risk it, the upside is negligible speedup. Also I would need something like concatMapWithIndex. :)

Reply:
javascript's max num size is (2^53 - 1), I would hope that we're not going to get a list of `DynamicProblem` that size!

Are you sure you need `concatMapWithIndex`? You can just convert a List to an Array and then `mapWithIndex` over it

Reply:
I see, I was assuming it was 32 bit. I still don't see the problem with using BigInt, but ok.

Reply:
I've replaced BigInt with int but I cannot use mapWithIndex because the function is not bijective. Surely I could filter before applying map, but that is just complicating things.